### ISSUE TITLE
DataLoader Error with ConcatDataset of MNIST and SVHN

### ISSUE THREAD

**Issue by wgchang** [pytorch/vision#595]
https://github.com/pytorch/vision/blob/152c2b3a9403811198beb8fffc5070fc80d1273c/torchvision/datasets/mnist.py#L70
### Discription
Hello. 
I tried to concatenate two datasets MNIST and SVHN.
Here, I found an error.
Example code:
```python
# Load MNIST
mnist_transform = transforms.Compose([
    transforms.ToTensor(),
])
mnist_test = toy_dataset.MNIST('./data/MNIST', train=False, download=True, transform=mnist_transform)
mnist_test_dataloader = data.DataLoader(mnist_test, batch_size=batch_size, shuffle=False,
                                        num_workers=args.num_workers, drop_last=False, pin_memory=True)
# Load SVHN
svhn_transform = transforms.Compose([
    transforms.Resize([28, 28]),
    transforms.Grayscale(),
    transforms.ToTensor()
])

svhn_test = toy_dataset.SVHN('./data/SVHN', split='test', download=True, transform=svhn_transform)
svhn_test_dataloader = data.DataLoader(svhn_test, batch_size=batch_size, shuffle=False,
                                       num_workers=args.num_workers, drop_last=False, pin_memory=True)
concat_dataset = mnist_test + svhn_test
concat_test_dataloader = data.DataLoader(concat_dataset, batch_size=batch_size, shuffle=False,
                                         num_workers=args.num_workers, drop_last=False)
#### ERROR #####
D = [d for d in merged_test_dataloader]
```
### Error Message
TypeError: expected Tensor as element 16 in argument 0, but got tuple

I simply put int() at line 70 for the target and solve this issue.
```python
img, target = self.data[index], int(self.targets[index])
```

**Comment by fmassa** [comment]
Thanks for catching this bug!
This is a bug in MNIST that happened when tensors and variables got merged and we started supporting scalar tensors.
Can you send a PR fixing this?

**Comment by fmassa** [comment]
Fixed via #605

**Comment by Oktai15** [comment]
Can we close this issue, @fmassa?
